app/models/search.rb

```diff
@@ -28,12 +28,16 @@ def filters
     self.class.filters
   end
 
-  validates :filter, presence: true, inclusion: {
-    in: filters.map { |f| f.id.to_s },
-    message: 'Filter "%{value}" is not recognized'
-  }
+  validates :filter,
+            presence: true,
+            inclusion: { in: filters.map { |f| f.id.to_s },
+                         message: 'Filter "%{value}" is not recognized' }
+
+  validates :term,
+            presence: true,
+            format: { with: /\A[A-Za-z0-9\s]+\z/ },
```

Comment:
Is there a potential issue here with names containing characters (like O'Reilly?)

Reply:
handle. have been talking to alan and testing. see last few commits - basically allow apostrophes but not hyphens (es handles these in unexpected ways and can timeout/error due to CDA issues.